// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTServerContract";
option optimize_for = LITE_RUNTIME;

import public "Enums.proto";
import public "ListenAddress.proto";
import public "Nym.proto";
import public "Signature.proto";

message ServerContract {
    optional uint32 version = 1;
    optional string id = 2;
    optional string nymid = 3;
    optional string name = 4;
    optional Nym publicNym = 5;
    repeated ListenAddress address = 6;
    optional string terms = 7;
    optional bytes transportKey = 8;
    optional Signature signature = 9;
}
